src/Response/Types/ErrorType.php

```diff
@@ -13,22 +16,209 @@ class ErrorType
+        array $jsonPointersInTemplate,
+        ResponseInterface $response
+    ): string {
+        if (count($jsonPointersInTemplate[0]) < 1) {
+            return $this->addPlaceHolderValue(
+                $errorDescription,
+                '{$response.body}',
+                $response->getRawBody()
+            );
+        }
+
+        $jsonResponsePointer = $this->initializeJsonPointer($response);
+
+        $jsonPointers = $jsonPointersInTemplate[0];
+
+        for ($x = 0; $x < count($jsonPointers); $x++) {
+            $placeHolderValue = "";
+
+            if ($jsonResponsePointer != null) {
+                $placeHolderValue = $this->getJsonPointerValue($jsonResponsePointer, ltrim($jsonPointers[$x], '#'));
+            }
+
+            $errorDescription = $this->addPlaceHolderValue(
+                $errorDescription,
+                '{$response.body' . $jsonPointers[$x] . '}',
+                $placeHolderValue
+            );
+        }
+
+        return $errorDescription;
+    }
+
+    private function getJsonPointersFromTemplate(string $template): array
+    {
+        $pointerPattern = '/#[\w\/]*/i';
+
+        preg_match_all($pointerPattern, $template, $matches);
+
+        return $matches;
+    }
+
+    private function addPlaceHolderValue(
+        string $template,
+        string $placeHolder,
+        $value,
+        bool $searchCaseInsensitive = false
+    ): string {
+        if (!is_string($value)) {
+            $value = var_export($value, true);
+        }
+
+        if ($searchCaseInsensitive) {
+            return str_ireplace($placeHolder, $value, $template);
+        }
+
+        return str_replace($placeHolder, $value, $template);
+    }
+
+    /**
+     * @param $headers array
+     * @return int[]|string[]
+     */
+    private function getHeaderKeys(array $headers)
+    {
+        return array_keys($headers);
+    }
+
+    /**
+     * @param $jsonPointer Pointer
+     * @param $pointer string
+     * @return mixed Json pointer value from the JSON provided.
+     */
+    private function getJsonPointerValue(Pointer $jsonPointer, string $pointer)
+    {
+        try {
+            if (trim($pointer) === '') {
+                return "";
+            }
+
+            $pointerValue = $jsonPointer->get($pointer);
+
+            if (is_object($pointerValue)) {
+                return CoreHelper::serialize($pointerValue);
+            }
+
+            return $pointerValue;
+        } catch (Pointer\NonexistentValueReferencedException | Pointer\InvalidPointerException $ex) {
+            return "";
+        }
+    }
+
+    /**
+     * @throws Pointer\InvalidJsonException
+     * @throws Pointer\NonWalkableJsonException
+     */
+    private function initializeJsonPointer(ResponseInterface $response): ?Pointer
+    {
+        $rawBody = $response->getRawBody();
+        $jsonResponsePointer = null;
+
+        if ($this->isJson($rawBody)) {
+            $jsonResponsePointer = new Pointer($rawBody);
```

Comment:
This line is throwing a warning that it may Throw exceptions, catch this exception and return null in catch block